src/validation.js

```diff
@@ -5,21 +5,19 @@ import { mapAsync } from './util/promise'
 export let defaultHasValue = x => !F.isBlankDeep(_.some)(x.data)
 // Aync fn to inspect types.
 // ASYNC runValidate: return true -> proceed, return false -> exclude, throw -> error!
-export let validate = _.curry(async (runValidate, extend, child) => {
-  delete child.error //?? might need to be on mutate only in case of server error?
+export let validate = _.curry(async (runValidate, child) => {
+  child.error = null
   try {
     if (child.children)
-      await mapAsync(validate(runValidate, extend), child.children)
+      await mapAsync(validate(runValidate), child.children)
     let hasValue = child.children
       ? _.some('hasValue', child.children)
       : await runValidate(child)
-    extend(child, { hasValue })
+    child.hasValue = hasValue
     return hasValue
   } catch (error) {
-    extend(child, {
```

Comment:
Again extend is better

Reply:
should we be using extend everytime we want to set a property on a node? we set a bunch of values in other functions without extend.